azavea/echo-locator
# Feature/kak/neighborhood routing#25

## Overview

 - Get the three route results for all neighborhoods for the set origin and time of day on map page load or input change and display in sidebar cards the three neighborhoods with the shortest best route.
 - Updates the data using the last provided spreadsheet, from March 13th.
 - Adds a workaround for a Firefox Leaflet rendering issue


### Demo

![image](https://user-images.githubusercontent.com/960264/54539961-7bdf3780-496d-11e9-90bc-dbe60b8be95f.png)

On time of day change:
![image](https://user-images.githubusercontent.com/960264/54539999-92858e80-496d-11e9-92ea-97efb99eb59a.png)


### Notes

The updated spreadsheet provided had a header row of column names that did not match the first spreadsheet we received. Here I renamed a couple of them in adding the CSV (one had a misspelling.) We will need to establish what the format of the neighborhood data should be and what fields to expect before we can read from the neighborhood GeoJSON properties extensively in the app.

Created #71 for adding non-transit travel modes. We need at least driving results support.

Created #72 for establishing an approach for using neighborhood data to rank them by metrics such as affordability and school ratings.

Created #73 for modifying the card layout to show neighborhood data.

The [Firefox rendering issue](https://github.com/Leaflet/Leaflet/issues/4686) surfaced when rendering the full list of neighborhoods in the sidebar, as the current styling will match the map height to the sidebar. 

There are some minor package updates available, which I'm postponing pulling in until [this router issue](https://github.com/ReactTraining/react-router/issues/6630) gets addressed.

I've left in the `Log` sidebar component for now, as it has been helpful for debugging.


## Testing Instructions

 * Create a profile and go to map page
 * Route results for the primary destination should show in sidebar
 * Changing the destination or time of day inputs should update results as expected
 * Page reload and in-app navigation should function as expected (load results for the set inputs)


Closes #25
